components/org.wso2.bfsi.consent.management.common/src/main/java/org/wso2/bfsi/consent/management/common/config/ConsentManagementConfigParser.java

```diff
@@ -0,0 +1,234 @@
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Stack;
+
+import javax.xml.stream.XMLStreamException;
+
+/**
+ * Config parser for bfsi-consent-management.xml.
+ */
+public class ConsentManagementConfigParser {
+
+    private static final Log log = LogFactory.getLog(ConsentManagementConfigParser.class);
+    // To enable attempted thread-safety using double-check locking
+    private static final Object lock = new Object();
+    private static volatile ConsentManagementConfigParser parser;
+    private static final Map<String, Object> configuration = new HashMap<>();
+
+    /**
+     * Private Constructor of config parser.
+     */
+    private ConsentManagementConfigParser() {
+
+        buildConfiguration();
+    }
+
+    /**
+     * Singleton getInstance method to create only one object.
+     *
+     * @return ConsentManagementConfigParser object
+     */
+    public static ConsentManagementConfigParser getInstance() {
+
+        if (parser == null) {
+            synchronized (lock) {
+                if (parser == null) {
+                    parser = new ConsentManagementConfigParser();
+                }
+            }
+        }
+        return parser;
+    }
+
+    /**
+     * Method to read the configuration (in a recursive manner) as a model and put them in the configuration map.
+     */
+    private void buildConfiguration() {
+
+        InputStream inStream = null;
+        StAXOMBuilder builder;
+        try {
+            File configXml = new File(CarbonUtils.getCarbonConfigDirPath(), ConsentManagementConstants.CONFIG_FILE);
+            if (configXml.exists()) {
+                inStream = new FileInputStream(configXml);
+            }
+            if (inStream == null) {
+                String message =
+                        "BFSI Consent Management configuration not found at: " + CarbonUtils.getCarbonConfigDirPath() +
+                                ConsentManagementConstants.CONFIG_FILE;
+                if (log.isDebugEnabled()) {
+                    log.debug(message.replaceAll("[\r\n]", ""));
+                }
+                throw new FileNotFoundException(message);
+            }
+            builder = new StAXOMBuilder(inStream);
+            OMElement rootElement = builder.getDocumentElement();
+            Stack<String> nameStack = new Stack<>();
+            readChildElements(rootElement, nameStack);
+        } catch (IOException | XMLStreamException | OMException e) {
+            throw new ConsentManagementRuntimeException("Error occurred while building configuration from" +
+                    " bfsi-consent-management.xml", e);
+        } finally {
+            try {
+                if (inStream != null) {
+                    inStream.close();
+                }
+            } catch (IOException e) {
+                log.error("Error closing the input stream for bfsi-consent-management.xml", e);
+            }
```

Comment:
Couldn't we handle this using try with resources?

Reply:
Fixed in https://github.com/wso2-extensions/bfsi-consent-management/pull/1/commits/2089256398b8b821f4bb347077d8da06d684d829